### PR TITLE
Switch to gcc-13 for C++23 tests

### PR DESCRIPTION
**Pull request by phprus**
But there are some build errors (``potential null pointer dereference [-Werror=null-dereference]``):

https://github.com/phprus/fmt/actions/runs/10708260895/job/29690153702

@vitaut

### CODE REVIEW

**Comment by vitaut** [comment]
Do you have an idea what pointer this warning refers to? `__location`?
```
/usr/include/c++/13/bits/stl_construct.h:97:14: error: potential null pointer dereference [-Werror=null-dereference]
   97 |     { return ::new((void*)__location) _Tp(std::forward<_Args>(__args)...); }
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by phprus** [comment]
This is gcc bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=108860

**Comment by phprus** [comment]
Errors from GoogleTest: https://github.com/fmtlib/fmt/actions/runs/10718554817/job/29720727961?pr=4155

**Comment by phprus** [comment]
@vitaut 
All errors is fixed.

File: test/gtest/gmock-gtest-all.cc [hunk 1]
```diff
@@ -13983,46 +13984,50 @@ MockObjectRegistry g_mock_object_registry;
 
 // Maps a mock object to the reaction Google Mock should have when an
 // uninteresting method is called.  Protected by g_gmock_mutex.
-std::map<const void*, internal::CallReaction> g_uninteresting_call_reaction;
+std::unordered_map<uintptr_t, internal::CallReaction>&
+UninterestingCallReactionMap() {
+  static auto* map = new std::unordered_map<uintptr_t, internal::CallReaction>;
+  return *map;
+}
```
Comment (vitaut) [inline test/gtest/gmock-gtest-all.cc:13991]: Is this a change from mainstream gmock?

Comment (vitaut) [reply test/gtest/gmock-gtest-all.cc:13991]: NVM, I see from the commit history that it is.

**Comment by vitaut** [comment]
Thank you!

File: test/gtest/gmock-gtest-all.cc (hunk 1 above)
Comment (phprus) [reply test/gtest/gmock-gtest-all.cc:13991]: Yes:

https://github.com/google/googletest/commit/0320f517fd920866d918e564105d68fd4362040a#diff-0dfa2d04b973dc072009e740a0c47e0524bd1650c1336470f70f3437c15a9488R540

I would suggest replacing it with:

```c++
std::unordered_map<uintptr_t, internal::CallReaction>&
UninterestingCallReactionMap() {
  static std::unordered_map<uintptr_t, internal::CallReaction> umap;
  return umap;
}
```

**Pull request merged**